### DISCUSSION TITLE
v 2.6.x Mac  build failure  maci64  - clang++  cmake build error . Doesnt support c++11

### DISCUSSION THREAD

**Discussion by ramcharan-palnati** [eProsima/Fast-DDS#3124] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Expected to build FastDDS on mac without any issues.

### Current behavior

CMake build failure of v2.6.x on mac  using clang++  compiler . Could you help me in  fixing this.

Build log : 

-- The C compiler identification is AppleClang 13.0.0.13000029
-- The CXX compiler identification is AppleClang 13.0.0.13000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: //devel/sandbox//3psandbox/3p/harness/scripts/macos_wrappers/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: //devel/sandbox//3psandbox/3p/harness/scripts/macos_wrappers/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Version: 2.6.1
-- Performing Test SUPPORTS_CXX20
-- Performing Test SUPPORTS_CXX20 - Failed
-- Performing Test SUPPORTS_CXX17
-- Performing Test SUPPORTS_CXX17 - Failed
-- Performing Test SUPPORTS_CXX14
-- Performing Test SUPPORTS_CXX14 - Failed
-- Performing Test SUPPORTS_CXX1Y
-- Performing Test SUPPORTS_CXX1Y - Failed
-- Performing Test SUPPORTS_CXX11
-- Performing Test SUPPORTS_CXX11 - Failed
CMake Error at cmake/common/check_configuration.cmake:83 (message):
  Force to support stdc++11 but not supported by the compiler
Call Stack (most recent call first):
  CMakeLists.txt:100 (check_stdcxx)

-- Configuring incomplete, errors occurred!

### Steps to reproduce

Build FastDDS v2.6.x  on mac 

### Fast DDS version/commit

v2.6.x

### Platform/Architecture

MacOS Mojave 10.14

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by ramcharan-palnati** [comment]
Could anyone help me with this? Is this expected while using clang?

**Comment by jingTian-z** [comment]
"Force to support stdc++11 but not supported by the compiler". This means that your compiler does not support C++11.
Also, the compiler you're using is "//devel/sandbox//3psandbox/3p/harness/scripts/macos_wrappers/cc", which looks a little weird.
Please check your compiler.

**Comment by Mario-DL** [comment]
Hi @ramcharan-palnati,

According to our [CONTRIBUTING.md](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md) guidelines, your compiler is not an officially supported one, so I am moving this issue to the corresponding discussion forum.
